ChronoVisor/src/VisorClientPortal.cpp

```diff
@@ -206,16 +206,14 @@ chronolog::VisorClientPortal::AcquireStory(chl::ClientId const &client_id, std::
     { return chronolog::AcquireStoryResponseMsg(CL_ERR_NOT_AUTHORIZED, story_id, recording_keepers); }
 
     int ret = CL_ERR_UNKNOWN;
-    //ret = chronicleMetaDirectory.create_story(chronicle_name, story_name, attrs);
-
-    //if (ret != CL_SUCCESS && ret != CL_ERR_STORY_EXISTS)
-    //{ return chronolog::AcquireStoryResponseMsg(ret, story_id, recording_keepers); }
 
     bool notify_keepers = false;
+
     ret = chronicleMetaDirectory.acquire_story(client_id, chronicle_name, story_name, attrs, flags, story_id
```

Comment:
should we delay the actual metadata operation after we successfully notify an active recording group? I know this change will be reversed if `notifyRecordingGroupOfStoryRecordingStart()` fails. but I think it would be better to not do it at all if something fails.
I know this might not be related to this PR. we may put a TODO here for later improvement.

Reply:
chronicleMetaDirectory.acquire_story() assigns story_id  to the story.
keeperRegistry.notifyRecordingGroupOfStoryRecordingStart() needs the story_id to communicate it to the recording keepers and grapher. 
this dictates the order of operations.

Reply:
> chronicleMetaDirectory.acquire_story() assigns story_id to the story. keeperRegistry.notifyRecordingGroupOfStoryRecordingStart() needs the story_id to communicate it to the recording keepers and grapher. this dictates the order of operations.

then we need to split the story_id generation to a new function and only call it before we notify any recording group. for sure this can be just TODO for this PR.

Reply:
agreed